contrib/endpoints/src/api_manager/check_security_rules.h

```diff
@@ -15,12 +15,84 @@
 #ifndef API_MANAGER_CHECK_SECURITY_RULES_H_
 #define API_MANAGER_CHECK_SECURITY_RULES_H_
 
-#include "contrib/endpoints/include/api_manager/utils/status.h"
+#include "contrib/endpoints/include/api_manager/api_manager.h"
+#include "contrib/endpoints/src/api_manager/auth/service_account_token.h"
 #include "contrib/endpoints/src/api_manager/context/request_context.h"
+#include "contrib/endpoints/include/api_manager/utils/status.h"
+
+#include <string>
+using ::google::api_manager::utils::Status;
 
 namespace google {
 namespace api_manager {
 
+class AuthzChecker : public std::enable_shared_from_this<AuthzChecker> {
+ public:
+  // Constructor
+  AuthzChecker(ApiManagerEnvInterface *env,
+               auth::ServiceAccountToken *sa_token);
+
+  // Check for Authorization success or failure
+  void Check(std::shared_ptr<context::RequestContext> context,
+             std::function<void(Status status)> continuation);
+
+ private:
+
+  // Helper method that invokes the test firebase service api.
+  void CheckAuth(std::string ruleset_id,
+                 std::shared_ptr<context::RequestContext> context,
+                 std::function<void(Status status)> continuation);
+
+  // Parse the respose for GET RELEASE API call
+  std::pair<Status, std::string> ParseReleaseResponse(std::string *json_str);
+
+  // Parses the response for the TEST API call
+  Status ParseTestResponse(std::shared_ptr<context::RequestContext> context,
+                           std::string *json_str);
+
+  // Builds the request body for the TESP API call.
+  std::string BuildTestRequestBody(
+      std::shared_ptr<context::RequestContext> context);
+
+  void AddToBody(const std::string &key,
```

Comment:
can some of these function static.  

if so,  they can be in the cc file, inside anonymous name space